pkg/endpoint/local_ip.go

```diff
@@ -68,6 +85,60 @@ func GetLocalIPForDestination(dst string, family k8snet.IPFamily) string {
 	return localIP
 }
 
+func isValidGlobalIPv6(ip net.IP) bool {
+	// fd00::/8 - avoid locally assigned ULA
+	if ip.To16() == nil || ip.IsLoopback() || ip.IsLinkLocalUnicast() ||
+		ip[0] == 0xfd {
+		return false
+	}
+	// Accept fc00::/8 or global unicast (2000::/3)
+	return ip[0] == 0xfc || (ip[0]&0xe0 == 0x20)
+}
+
+func getValidGlobalIPv6FromSameInterface(ip net.IP) (string, error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return "", fmt.Errorf("failed to list interfaces: %w", err)
+	}
+
+	for _, iface := range interfaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			logger.Warningf("Failed to list the address from the interface %v", iface.Name)
+			continue
+		}
+
+		// Confirm this interface owns the original (non-global) IP
+		ownsIP := false
+
+		for _, addr := range addrs {
+			if ipNet, ok := addr.(*net.IPNet); ok && ipNet.Contains(ip) {
```

Comment:
Also we can skip non-IPv6 addresses , right ?

Reply:
Done.